shared/src/metabase/mbql/schema.cljc

```diff
@@ -78,6 +79,14 @@
    (apply s/enum datetime-bucketing-units)
    "datetime-bucketing-unit"))
 
+(def TimezoneId
+  "Valid timezone id."
+  #?(:clj  (s/named
+             (apply s/enum (ZoneId/getAvailableZoneIds)) ;; 600 timezones on java 17
+             "timezone-id")
+     ;; TODO figure out how to use moment.tz.names() or Intl.supportedValuesOf('timeZone') cljs
+     :cljs (s/named helpers/NonBlankString "timezone-id")))
```

Comment:
I'm experimenting at the shadow node repl

```clojure
❯ npx shadow-cljs node-repl
shadow-cljs - config: /Users/dan/projects/work/metabase/shadow-cljs.edn
shadow-cljs - connected to server
> (require '["moment" :as m] '["moment-timezone" :as mtz])
nil
cljs.user=>  (take 4 (.names (.-tz moment)))
("Africa/Abidjan" "Africa/Accra" "Africa/Addis_Ababa" "Africa/Algiers")
cljs.user=>
```

Why the reversion back to just non-blank strings?

Reply:
Run `yarn run test-cljs` will failed to compile.

```
------ WARNING #1 - :undeclared-var --------------------------------------------
 File: /Users/earther/metabase/metabase/shared/src/metabase/mbql/schema.cljc:90:42
--------------------------------------------------------------------------------
  87 |              "timezone-id")
  88 |      ;; TODO figure out how to use moment.tz.names() or Intl.supportedValuesOf('timeZone') cljs
  89 |      :cljs (s/named
  90 |              (apply s/enum (.names (.-tz moment))) ;; 600 timezones on java 17
------------------------------------------------^-------------------------------
 Use of undeclared Var metabase.mbql.schema/moment
--------------------------------------------------------------------------------
  91 |              "timezone-id")))
  92 |
  93 | (def TemporalExtractUnits
  94 |   "Valid units to extract from a temporal."
--------------------------------------------------------------------------------

SHADOW import error /Users/earther/metabase/metabase/.shadow-cljs/builds/test/dev/out/cljs-runtime/metabase.mbql.schema.js

/Users/earther/metabase/metabase/target/node-tests.js:64
  /* ignore this, look at stacktrace */ fn.call(global, require, module, __filename, __dirname);
                                           ^
TypeError: Cannot read properties of undefined (reading 'tz')                                      
```